src/backend/CommandBufferStateTracker.cpp

```diff
@@ -274,6 +270,10 @@ namespace backend {
     }
 
     bool CommandBufferStateTracker::SetBindGroup(uint32_t index, BindGroupBase* bindgroup) {
+        if (!HavePipeline()) {
```

Comment:
nit: I'm not sure we want to add this constraint. It doesn't help any backend (Metal and Vulkan don't care, D3D12 already needs to reset bindgroups on any layout change).